### ISSUE TITLE
"Warn before quitting" should be off by default

### ISSUE THREAD

**Issue by domoritz** [vector-im/element-web#17311]
On MacOS, when running CMD+Q, I would expect Element to close but I need to first confirm that I want to close the app. I believe the preference "Warn before quitting" should be off by default just like any other app where you don't lose anything when you close the app.

Maybe it's already the default but I couldn't find anything in the code or the docs.

**Comment by t3chguy** [comment]
> should be off by default just like any other app where you don't lose anything when you close the app.

Its not off by default in Chrome on MacOS

**Comment by domoritz** [comment]
Good point. However, in a browser people may actually lose something they care about (e.g. open tabs, something they typed into a form etc).

Element is an application and a reversible action should not require a confirmation prompt by default.

**Comment by t3chguy** [comment]
Webpages have a `beforeunload` handler which can trigger a dialog for that situation, and that'll be shown regardless of `Warn Before Quitting` setting

> Element is an application and a reversible action should not require a confirmation prompt by default.

Not everything you do in Element is lossless, e.g if you're mid-way through changing the topic of a room, when you restart the app you'll be back on the room timeline, settings closed, progress lost

**Comment by domoritz** [comment]
> Webpages have a beforeunload handler which can trigger a dialog for that situation, and that'll be shown regardless of Warn Before Quitting setting

Excellent idea. Element should implement that for cases when the user is in the middle of making a non-recoverable edit or has unsent messages. However, I strongly believe that the confirmation should not be required 100% of the time.

**Comment by t3chguy** [comment]
It'd be really difficult to identify and implement for each and every state which is not fully recoverable.

> Element should implement that for cases when the user is in the middle of making a non-recoverable edit or has unsent messages.

Edits, unsent messages, drafts should all already be persisted. Other things like state of filters, other form elements, will not be.

**Comment by domoritz** [comment]
I see. I can reword the issue then as a feature request to work towards not requiring confirmation for quitting the app.

**Comment by t3chguy** [comment]
I mean, this was a feature that was recently added by user request, it has a user preference for a reason, whether or not that's on/off by default seems very minimal and open to one's own opinion.

**Comment by aaronraimist** [comment]
Would be nice if the warn before quit dialog had a "close and don't warn me again" button so you don't have to look around to find the specific preference.

**Comment by domoritz** [comment]
The right solution is to only show a dialog when there is some state that would get lost and not even have a toggle in the settings.
